### PR TITLE
AST-2860 - Read Time option added to Meta options

### PR DESCRIPTION
**Pull request by neerajmasai**
### Description

- Added Read Time option to meta options.
- JIRA - https://brainstormforce.atlassian.net/browse/AST-2860

### Screenshots

https://d.pr/v/DgsgKz

### Types of changes

- bugfix - Read Time option missing from Single Post & CPT
- Modified dynamic single post structure's meta config to include Read Time option.
- Added backwards compatibility.

### How has this been tested?

- Customizer > Single Posts > Post title > Meta options.
- Read Time option should be visible & functional.
- Check for other post types (Pages, CPT's)
- Check for backwards compatibility.

### Checklist:
- [x] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/theme-update/astra-update-functions.php [hunk 1]
```diff
@@ -838,6 +838,9 @@ function astra_theme_background_updater_4_0_0() {
 								$tax_slug    = 'ast-dynamic-single-' . esc_attr( $post_type ) . '-taxonomy-' . $tax_counter;
 							}
 							break;
+						case 'read-time':
+							$migrated_post_metadata[] = 'read-time';
+							break;
```
Comment (imnavanath) [inline inc/theme-update/astra-update-functions.php:843]: @neerajmasai will this code run for next versions?

Comment (neerajmasai) [reply inc/theme-update/astra-update-functions.php:843]: @imnavanath 
According to my thinking when old user updates from previous versions to 4.0 the code will run, but yes right it won't work for current users updating to latest Astra version. In that case, I think I need to add an update function for current version and add the backward flag accordingly for read-time, please correct me if I am wrong.

Comment (imnavanath) [reply inc/theme-update/astra-update-functions.php:843]: Let's do the changes as discussed.

File: inc/modules/posts-structures/customizer/class-astra-posts-single-structures-configs.php [hunk 2]
```diff
@@ -263,6 +263,12 @@ public function register_configuration( $configurations, $wp_customize ) {
 				);
 			}
 
+			// Display Read Time option in Meta options only when Astra Addon is activated.
+			/** @psalm-suppress UndefinedClass */ // phpcs:ignore Generic.Commenting.DocComment.MissingShort
+			if ( defined( 'ASTRA_EXT_VER' ) && Astra_Ext_Extension::is_active( 'blog-pro' ) ) {
+				$meta_config_options['read-time'] = __( 'Read Time', 'astra-addon' );
```
Comment (imnavanath) [inline inc/modules/posts-structures/customizer/class-astra-posts-single-structures-configs.php:269]:
```suggestion
				$meta_config_options['read-time'] = __( 'Read Time', 'astra' );
```

File: inc/theme-update/astra-update-functions.php [hunk 3]
```diff
@@ -991,3 +991,21 @@ function astra_theme_background_updater_4_0_0() {
...
+ *
+ * 1. Read Time option backwards handling for old users.
+ *
+ * @since x.x.x
+ * @return void
+ */
+function astra_theme_background_updater_4_0_2() {
+	$theme_options = get_option( 'astra-settings', array() );
+	if ( isset( $theme_options['blog-single-meta'] ) && in_array( 'read-time', $theme_options['blog-single-meta'] ) ) {
+		if( ! isset( $theme_options[ 'ast-dynamic-single-post-metadata' ] ) || ! in_array( 'read-time', $theme_options[ 'ast-dynamic-single-post-metadata' ] ) ) {
```
Comment (imnavanath) [inline inc/theme-update/astra-update-functions.php:1006]:
```suggestion
		if( is_array( $theme_options[ 'ast-dynamic-single-post-metadata' ] ) || ! in_array( 'read-time', $theme_options[ 'ast-dynamic-single-post-metadata' ] ) ) {
```

File: changelog.txt [hunk 4]
```diff
@@ -3,6 +3,7 @@ v4.0.2 (Unreleased)
 - Fix: exclude_from_search query parameter removed from get_post_types list to get CPTs under Customizer > Custom Post Types  section, case of CartFlows Step custom post type.
 - Fix: Archive - Title area 2 - Archive type title gets added as prefix on archive pages.
 - Fix: Text transformed to 'capitalize' for single post titles & archive titles after 4.0.0.
+- Fix: Read Time option missing from Single Post.
```
Comment (imnavanath) [inline changelog.txt:6]:
```suggestion
- Fix: Single Post - "Read Time" option missing from metadata.
```

File: inc/theme-update/astra-update-functions.php [hunk 5]
```diff
@@ -991,3 +991,22 @@ function astra_theme_background_updater_4_0_0() {
...
+ *
+ * 1. Read Time option backwards handling for old users.
+ *
+ * @since x.x.x
+ * @return void
+ */
+function astra_theme_background_updater_4_0_2() {
+	$theme_options = get_option( 'astra-settings', array() );
+	if ( isset( $theme_options['blog-single-meta'] ) && in_array( 'read-time', $theme_options['blog-single-meta'] ) ) {
+		if( ! isset( $theme_options['v4.0.2-flag'] ) && isset( $theme_options['ast-dynamic-single-post-metadata'] ) && ! in_array( 'read-time', $theme_options['ast-dynamic-single-post-metadata'] ) ) {
```
Comment (imnavanath) [inline inc/theme-update/astra-update-functions.php:1006]:
```suggestion
		if( isset( $theme_options['ast-dynamic-single-post-metadata'] ) && ! in_array( 'read-time', $theme_options['ast-dynamic-single-post-metadata'] ) ) {
```

File: inc/theme-update/astra-update-functions.php [hunk 6]
```diff
@@ -991,3 +991,22 @@ function astra_theme_background_updater_4_0_0() {
...
+ * 4.0.2 backward handling part.
+ *
+ * 1. Read Time option backwards handling for old users.
+ *
+ * @since x.x.x
+ * @return void
+ */
+function astra_theme_background_updater_4_0_2() {
+	$theme_options = get_option( 'astra-settings', array() );
+	if ( isset( $theme_options['blog-single-meta'] ) && in_array( 'read-time', $theme_options['blog-single-meta'] ) ) {
```
Comment (imnavanath) [inline inc/theme-update/astra-update-functions.php:1005]:
```suggestion
	if ( ! isset( $theme_options['v4.0.2-flag'] ) && isset( $theme_options['blog-single-meta'] ) && in_array( 'read-time', $theme_options['blog-single-meta'] ) ) {
```

File: inc/theme-update/astra-update-functions.php [hunk 7]
```diff
@@ -991,3 +991,22 @@ function astra_theme_background_updater_4_0_0() {
...
+		if( ! isset( $theme_options['v4.0.2-flag'] ) && isset( $theme_options['ast-dynamic-single-post-metadata'] ) && ! in_array( 'read-time', $theme_options['ast-dynamic-single-post-metadata'] ) ) {
+			$theme_options['ast-dynamic-single-post-metadata'][] = 'read-time';
+			$theme_options['v4.0.2-flag'] = false;
```
Comment (imnavanath) [inline inc/theme-update/astra-update-functions.php:1008]:
```suggestion
			$theme_options['v4-0-2-update-migration'] = true;
```

Comment (imnavanath) [reply inc/theme-update/astra-update-functions.php:1008]: Update flag likewise on top.

**Review by imnavanath — DISMISSED** [review]

**Pull request merged**